.github/workflows/snyk.yml

```diff
@@ -17,16 +17,21 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-python@v4
         with:
-          python-version: "3.9"
+          python-version: "3.9.12"
+          cache: "pip"
+          cache-dependency-path: engine/requirements.txt
       - uses: actions/setup-node@v3
         with:
           node-version: 14.17.0
+          cache: "yarn"
+          cache-dependency-path: grafana-plugin/yarn.lock
       - uses: snyk/actions/setup@master
-      - name: Install Dependencies
-        run: |
-          pip install -r engine/requirements.txt
-          cd grafana-plugin/
-          yarn --network-timeout 500000
+      - name: Install backend dependencies
+        working-directory: engine
+        run: pip install -r requirements.txt
+      - name: Install frontend dependencies
+        working-directory: grafana-plugin
+        run: yarn install --frozen-lockfile --prefer-offline --network-timeout 500000
```

Comment:
`--prefer-offline`

Interesting, does this work on CI or it will still attempt to download?